docs/gosec.md

```diff
@@ -0,0 +1,42 @@
+# GOSEC static analysis 
+## Background
+Gosec (https://github.com/securego/gosec) is a static code analysis tool that scans the code to find security related issues.
+
+Gosec executes a set of validation tests/rules and produces a report listing all the detected potential violations. The set of rules can be configured as well as include and exclude directories. 
+
+## Running gosec in your local environment
+To perform gosec scanning over your code run:
+```
+    make gosec
+```
+This command will execute all the tests. If you want to run a specific gosec test you can use the `GOSEC` environment variable: E.g.,  `GOSEC="G601`"
+
+## Handle false positive detected issues
+As a static analysis tool, gosec may produce false positive warnings - identify a potential risk that is not relevant/applicable in the specific KubeVirt code. 
+In extreme cases you can annotate the code with a special comment:
+```
+// #nosec <description>
+```
+This will instruct gosec to suppress the warning.
+The developer is encourage to avoid such annotation when possible as this will silence the warnnings. 
+### Fixing guidelines
+Define a new “unsafe” function that encapsulate the risky function or code
+The unsafe function name should give good indication on when this function should be used. Whenever possible, the function should validate in-code that the condition really applies.
+Annotate the call with a `\\#nosec ` comment only within this unsafe function implementation
+
+### Example  
+The rule “G204: Subprocess launched with variable” aimed to validate there is no risk of command or parameters injection from an attacker.
+For example, this rule will produce a warning every time the command `exec.Command(cmd)`, is used with a non fixed string `cmd`.
+However, in many occasions, while the string is not fixed it has no injection risk.
+
+**Suggested fix** 
+Define a new “unsafe” version of the Command() function and encapsulate the original `exec.Command(cmd)` in it.
+```
+// This function should be used only in case all parameters 
+// were validated against a possible attacker injection
+func ExecCommandWithNonInjectableParameters(cmd string) *Cmd {
```

Comment:
Just to clarify this: We assume that with this explicit function name people are more aware of potential risks and we hope that they are more careful? Very often we pass through some variables through layers. So I am not sure if someone changing this code may ever see the actual fuction call.

Is there something like a gosec mode which tries to not reports false positives? We use for instance `nogo` already which uses by default only checks and linters which try to only report real positives.

Reply:
There is no option of not reporting "false positive" as this is the definition of "false positive" :-)  
There is an option to disable all warnings for the rule, but here we are talking about rules we want to run. 

Yes, the assumption is that the name indicates the potential risk. This should help us against changes in the immediate location of that function, if there are side effects (i.e., a parameter that is used in this new function is passed through several layers and might changed there), indeed the name of the function won't help, however would claim that this is case should probably be fixed rather then changed to use "unsafe function".
For example - if the parameter to exec is going through several layer you can't really guarantee it's safe against injection, and you probably need to construct that parameter if a safer way.

Reply:
right, and I think you should put this in your text: code authors and reviewers should take great measures to ensure that the unsafe function is used safely, and would always be used safely. The painfully long name should server as a deterrent from over usage.

Reply:
> There is no option of not reporting "false positive" as this is the definition of "false positive" :-)

Sure, but still many tools have options to keep the false positive rate very-very low to stay relevant.

> For example - if the parameter to exec is going through several layer you can't really guarantee it's safe against injection, and you probably need to construct that parameter in a safer

It is not clear to me on how to create it in a save way to not trigger a gosec warning. As soon as I take e.g. a struct member into account, wouldn't it always be "unsafe"? Now with this approach I can just change on the very outside of the app what I pass in, and the call inside a struct member would still be considered save. For sure I have to validate things upfront there, but this internal check is not helpful at all. We would initially silence it and from that point on things can rot like before and become unsafe.

Reply:
@rmohr I tend to agree with you on this specific rule.  There is a lot of criticism against that rule as there is really no way  to fix it to avoid this warning. We will probably just use that rule as optional for audit/review warnings and not as PR gating..
I should find a better example.